File: programs/test/module-06/lesson-02/step-08/main.c
// module6 -> lesson2 -> step8
// Проверка знаков частного и остатка двух чисел при делении


#include <stdio.h>

int main()
{
    int x, y;
    scanf("%d%d", &x, &y);

    printf("%2d : %2d = %2d (остаток %2d)\n",  x,  y,   x  /   y,    x  %   y );
    printf("%2d : %2d = %2d (остаток %2d)\n",  x, -y,   x  / (-y),   x  % (-y));
    printf("%2d : %2d = %2d (остаток %2d)\n", -x,  y, (-x) /   y,  (-x) %   y );
    printf("%2d : %2d = %2d (остаток %2d)\n", -x, -y, (-x) / (-y), (-x) % (-y));
    return 0;
}

File: programs/test/module-06/lesson-02/step-08/main.py
# module6 -> lesson2 -> step8
# Проверка знаков частного и остатка двух чисел при делении

a = int(input())
b = int(input())

print(f' {a} //  {b} =  {  a  //   b } (остаток  {  a  %   b })')
print(f' {a} // {-b} = {   a  // (-b)} (остаток {   a  % (-b)})')
print(f'{-a} //  {b} = { (-a) //   b } (остаток  {(-a) %   b })')
print(f'{-a} // {-b} =  {(-a) // (-b)} (остаток { (-a) % (-b)})')
